### PR TITLE
websocket improvements

### PR DESCRIPTION
**Pull request by 2is10**

### CODE REVIEW

File: packrat/scripts/lib/reconnecting-websocket.js [hunk 1]
```diff
@@ -49,7 +46,25 @@ function ReconnectingWebSocket(url, onmessage) {
       }
     };
 
-    ws.onmessage = onmessage.bind(self);
+    ws.onmessage = function(e) {
+      if (e.data === '["hi"]') {
```
Comment (gmethvin) [inline packrat/scripts/lib/reconnecting-websocket.js]: it seems a little kludgy to compare to a string here instead of parsing it as JSON.

Comment (2is10) [inline packrat/scripts/lib/reconnecting-websocket.js]: I was going for smaller / faster / clearer.

Are you concerned that the server might add unnecessary whitespace, which would break the comparison?

Comment (2is10) [inline packrat/scripts/lib/reconnecting-websocket.js]: I'd actually prefer not using JSON for these special signal messages, but the `WebSocket` is declared to only return `JsArray` in `ExtStreamController`, since that's what application-level messages use. I do like the type-safety for them, so I went this route.
